src/extension.ts

```diff
@@ -672,11 +691,20 @@ function continueEditing(outline: DocumentOutline, useContext: boolean, selectio
 			);
 		}).then(offsetDiff => {
 			completion.translateAnchorOffset(offsetDiff);
-			return completion.completeText(
-				chatMessageBuilder.toChatMessages(),
-				userEndLineEol,
-				chatMessageBuilder.getCopilotOptions(),
-			);
+			if (errorMessage !== null) {
+				// Show errors to help users correct their text
+				completion.insertText(
+					errorMessage,
+					userEndLineEol
+				);
+			} else {
+				completion.completeText(
+					chatMessageBuilder.toChatMessages(),
+					userEndLineEol,
+					chatMessageBuilder.getCopilotOptions(),
+				);
+			}
+
```

Comment:
Now Copilot inserts error messages into editor buffer. Should we use the code from line 708 instead, since they are similar?

Reply:
Your code is good.
We will probably clean it up in a future refactoring.